docs/devenv.rst

```diff
@@ -0,0 +1,593 @@
+    /home/user/mopidy-dev/mopidy
+    .%
+    $
+
+It will also create a ``mopidy`` executable inside the virtualenv that will
+always run the latest code from the Git repo. Using another
+virtualenvwrapper command, ``cdvirtualenv``, we can show that too::
+
+    $ cdvirtualenv
+    $ cat bin/mopidy
+    ...
+
+The executable should contain something like this, using :mod:`pkg_resources`
+to look up Mopidy's "console script" entry point::
+
+    #!/home/user/virtualenvs/mopidy/bin/python2
+    # EASY-INSTALL-ENTRY-SCRIPT: 'Mopidy==0.19.5','console_scripts','mopidy'
+    __requires__ = 'Mopidy==0.19.5'
+    import sys
+    from pkg_resources import load_entry_point
+
+    if __name__ == '__main__':
+        sys.exit(
+            load_entry_point('Mopidy==0.19.5', 'console_scripts', 'mopidy')()
+        )
+
+.. note::
+
+    It still works to run ``python mopidy`` directly on the
+    :file:`~/mopidy-dev/mopidy/mopidy/` Python package directory, but if
+    you don't run the ``pip install`` command above, the extensions bundled
+    with Mopidy will not be registered with :mod:`pkg_resources`, making Mopidy
+    quite useless.
+
+Third, the ``pip install`` command will register the bundled Mopidy
+extensions so that Mopidy may find them through :mod:`pkg_resources`. The
+result of this can be seen in the Git repo, in a new directory called
+:file:`Mopidy.egg-info`, which is ignored by Git. The
+:file:`Mopidy.egg-info/entry_points.txt` file is of special interest as it
+shows both how the above executable and the bundled extensions are connected to
+the Mopidy source code:
+
+.. code-block:: ini
+
+    [console_scripts]
+    mopidy = mopidy.__main__:main
+
+    [mopidy.ext]
+    http = mopidy.http:Extension
+    local = mopidy.local:Extension
+    mpd = mopidy.mpd:Extension
+    softwaremixer = mopidy.softwaremixer:Extension
+    stream = mopidy.stream:Extension
+
+.. warning::
+
+   It's not uncommon to clean up in the Git repo now and then, e.g. by running
+   ``git clean``.
+
+   If you do this, then the :file:`Mopidy.egg-info` directory will be removed,
+   and :mod:`pkg_resources` will no longer know how to locate the "console
+   script" entry point or the bundled Mopidy extensions.
+
+   The fix is simply to run the install command again::
+
+       pip install --editable .
+
+Finally, we can go back to the workspace, again using a virtualenvwrapper
+tool::
+
+   cdproject
+
+
+.. _running-from-git:
+
+Running Mopidy from Git
+=======================
+
+As long as the virtualenv is activated, you can start Mopidy from any
+directory. Simply run::
+
+    mopidy
+
+To stop it again, press :kbd:`Ctrl+C`.
+
+Every time you change code in Mopidy or an extension and want to see it
+live, you must restart Mopidy.
+
+If you want to iterate quickly while developing, it may sound a bit tedious to
+restart Mopidy for every minor change. Then it's useful to have tests to
+exercise your code...
+
+
+.. _running-tests:
+
+Running tests
```

Comment:
Is this still an issue after switching to pytest? Due to experience from work, I suspect it does something different and smarter here. Maybe leave this out for now and revisit it if you continue to experience it with pytest.